### PR TITLE
[python] drop use of fontTools.misc.py23

### PR DESCRIPTION
**Pull request by josh-hadley**
- numerous modules
- also simplified `fdkutils.py` routines that do shell output (see [#847](https://github.com/adobe-type-tools/afdko/issues/847))

### CODE REVIEW

File: python/afdko/autohint.py [hunk 1]
```diff
@@ -1303,11 +1305,15 @@ def hintFile(options):
 					logMsg("No new hints. All glyphs were already hinted.")
 			else:
 				logMsg("No glyphs were hinted.")
+
 	if options.usePlistFile and (anyGlyphChanged or pListChanged):
 		#  save font plist file.
-		fontPlist.write(fontPlistFilePath)
+		with open(fontPlistFilePath, 'wb') as f:
+			plistlib.writePlist(fontPlist, f)
```
Comment (miguelsousa) [inline python/afdko/autohint.py]: use `dump()` instead?

Comment (josh-hadley) [reply python/afdko/autohint.py]: Eh. If we were planning to keep `autohint.py` current I guess that would be the right thing to do (actually, the right thing would be to use `fontTools.misc.plistlib`)

Comment (miguelsousa) [reply python/afdko/autohint.py]: only suggested it because you used `plistlib.load()` upstream

Comment (josh-hadley) [reply python/afdko/autohint.py]: Indeed. I'll change it to `dump()` in next round of changes.

**Pull request merged**